.gitignore

```diff
@@ -1 +1,129 @@
-TODO
```

Comment:
The intention here is that all this belongs in your global gitignore, since all this stuff should be ignored across all repos.